api/src/main/java/org/apache/iceberg/actions/RewriteManifests.java

```diff
@@ -56,6 +57,26 @@
    */
   RewriteManifests stagingLocation(String stagingLocation);
 
+
+  /**
+   * Allows reading of data files to repair them.
+   * @param mode repair mode
+   * @return this for method chaining
+   */
+  RewriteManifests repair(RepairMode mode);
```

Comment:
I generally prefer to use configuration methods rather than enums in public APIs. That would also allow us to avoid the NONE option and have the actual repair be a bit more specific, like `repairFileLengths()`. What do you guys think, @szehon-ho, @RussellSpitzer?